File: lol/src/test/kotlin/com/your/lol/support/GlobalTestUtils.java
package com.your.lol.support;

import com.google.gson.reflect.TypeToken;
import com.your.lol.dto.riot.MatchDto;
import com.your.lol.dto.statistics.ChampionDataDto;
import com.your.lol.dto.statistics.StatsDto;
import org.jetbrains.annotations.NotNull;
import org.junit.jupiter.api.function.Executable;

import java.util.List;

import static org.assertj.core.api.Assertions.assertThat;
import static org.assertj.core.api.Assertions.offset;

public abstract class GlobalTestUtils {

    @NotNull
    protected static <T> Executable assertEquals(T actual, T expected) {
        return () -> assertThat(actual).isEqualTo(expected);
    }

    @NotNull
    protected static Executable assertCloseTo(double actual, double expected, double offset) {
        return () -> assertThat(actual).isCloseTo(expected, offset(offset));
    }

    @NotNull
    protected static ChampionDataDto findChampionByName(StatsDto stats, String championName) {
        return stats.getChampionDataDtos().stream()
                .filter(it -> it.getChampionName().equals(championName))
                .findAny()
                .get();
    }

    protected List<MatchDto> createTestData(String filePath) {
        return JsonFileConverter.fromJsonFile(
                filePath,
                new TypeToken<>() {
                });
    }
}

File: lol/src/test/kotlin/com/your/lol/support/GlobalTestUtils.kt
package com.your.lol.support

import com.google.gson.reflect.TypeToken
import com.your.lol.dto.riot.MatchDto
import com.your.lol.dto.statistics.ChampionDataDto
import com.your.lol.dto.statistics.StatsDto
import org.assertj.core.api.Assertions
import org.assertj.core.data.Offset.offset
import org.junit.jupiter.api.function.Executable


abstract class GlobalTestUtils {

    protected fun <T> assertEquals(actual: T, expected: T): Executable {
        return Executable { Assertions.assertThat(actual).isEqualTo(expected) }
    }

    protected fun <T> assertCloseTo(actual: Double, expected: Double, offset: Double): Executable {
        return Executable { Assertions.assertThat(actual).isCloseTo(expected, offset(offset)) }
    }

    protected fun findChampionByName(stats: StatsDto, championName: String): ChampionDataDto {
        return stats.championDataDtos!!.stream()
                .filter { it.championName == championName }
                .findAny()
                .get()
    }

    protected fun createTestData(filePath: String): List<MatchDto> {
        return JsonFileConverter.fromJsonFile(filePath, object : TypeToken<List<MatchDto>>() {})
    }
}